analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Binary.cs

```diff
@@ -44,6 +46,52 @@ protected override ProgramState LearnBranchingConstraint(ProgramState state, IBi
         return state;
     }
 
+    protected override ProgramState PreProcess(ProgramState state, IBinaryOperationWrapper operation)
+    {
+        if (operation.LeftOperand is { } leftOperand
+            && operation.RightOperand is { } rightOperand
+            && state[leftOperand]?.Constraint<NumberConstraint>() is { } leftNumber
+            && state[rightOperand]?.Constraint<NumberConstraint>() is { } rightNumber
+            && Calculate(operation.OperatorKind, leftNumber, rightNumber) is { } newConstraint)
+        {
+            state = state.SetOperationConstraint(operation.WrappedOperation, newConstraint);
+        }
+        return state;
+    }
+
+    private static NumberConstraint Calculate(BinaryOperatorKind kind, NumberConstraint left, NumberConstraint right)
+    {
+        BigInteger? min, max;
+        switch (kind)
+        {
+            case BinaryOperatorKind.Add:
+                min = Calculation(kind, left.Min, right.Min);
+                max = Calculation(kind, left.Max, right.Max);
+                break;
+            case BinaryOperatorKind.Subtract:
+                min = Calculation(kind, left.Min, right.Max);
+                max = Calculation(kind, left.Max, right.Min);
```

Comment:
Note for myself: I need to take one more look at this logic once the indirection is out of the way

Reply:
Maybe this helps:
https://sharplab.io/#v2:C4LgTgrgdgNAJiA1AHwG4EMwAIC2BLKARiwF4tCBuAWACgNsd0APYswgBmrs1wICZSWPl3q5mAsgFYRPfFACyBCAGdFUQXOIBaXlGG1RctSvnMNBbWKb7uDZsdVmyjFlh0ubh+0tUENzSzkbWk1BShCAsM4Q/kEtaQjrOJsjHzVzIjddFIIHUyYMwPEuFzynKyLrEu8oEz9nSPd+LhjMsi0Oatd28JogwRTxQQS+3LT63UDm1rKC5wssjy7Z/1d3YsS8iZcpvRbRtrdOxNYjktj2kY9k84Vx9XnMpr2ZtPLQ9arNt7mKxY2+jU6g8/s9ggdTh1ooDuuRof1Ll0JFgrmNar4QR9srcVo9KoMmLjQVYCVtMY1sUA==